utils/build-script-impl

```diff
@@ -1756,7 +1768,8 @@ for host in "${ALL_HOSTS[@]}"; do
                     )
                 fi
 
-                if [[ ! "${SKIP_BUILD_ANDROID}" ]]; then
+                if [[ ! "${SKIP_BUILD_ANDROID}" ]] ||
+                   [[ $(is_cross_tools_host ${host}) && "${host}" == "android-"* ]]; then
```

Comment:
Is this because if you don't skip building Android, the stdlib would also be built? Isn't there a cleaner or clearer way of doing this?

Reply:
No, this is a minor addition to signify that the NDK can also be used to cross-compile the host tools and not just the Android stdlib, as [`SKIP_BUILD_ANDROID` confusingly just means "don't build the Android stdlib."](https://github.com/apple/swift/blob/8fc259816734e98177082cd8ea6045bc1460bc9f/utils/build_swift/build_swift/driver_arguments.py#L962) It fixes a bug I hit where I was only cross-compiling the host tools for Android and _not_ the stdlib, which the unmodified version of this check would screw up.

Reply:
I don't believe that any of the Swift host tools, like the Swift compiler or SourceKit, use any Swift yet, as I was building the host tools alone till earlier this year. A search for `.swift` files in the `lib/` and `tools/` directories only turns up anything in `tools/swift-inspect`, which I don't think is built by default.